src/mococrw/padding_mode.h

```diff
@@ -79,6 +80,9 @@ class NoPadding: public RSAPadding {
     */
     int getDataMaxSize(const AsymmetricPublicKey& key) const override
     {
+        if(key.getType() != AsymmetricKey::KeyTypes::RSA){
```

Comment:
I wonder if having two derived classes, (AsymmetricKeyRSA and AssymetricKeyEC, for example) would be better in the sense that having these guards would no longer be necessary

Reply:
Backwards compatibility is indeed a problem, but these kind of guards are a bit inelegant in my opinion.

Reply:
Most of the guards will be removed once we add support for ECC signatures and certificates.
Only the encryption functionality will need them

Reply:
Removed some of the guards by replacing RSA_size with EVP_PKEY_size